lib/encoder.py

```diff
@@ -59,6 +59,14 @@ def __init__(self, name, values=[]):
     # pylint: disable=W0102
     self.name = name
     self.values = frozenset(values)
+    self.mandatory = False
+
+  def Mandatory(self):
```

Comment:
I think it's the strategy used in the builder pattern, but not sure it's limited to that, or named after that.